### ISSUE TITLE
Build fails on FreeBSD: undefined: syscall.Mprotect

### ISSUE THREAD

**Issue by yurivict** [wasilibs/go-re2#110]
While building trufflehog-3.78.1 - these errors occur:
```
# github.com/wasilibs/go-re2/internal/alloc
vendor/github.com/wasilibs/go-re2/internal/alloc/alloc.go:58:16: undefined: syscall.Mprotect
vendor/github.com/wasilibs/go-re2/internal/alloc/alloc.go:80:18: undefined: syscall.Mprotect
```

Go-1.22
FreeBSD 14.1

**Comment by anuraaga** [comment]
@yurivict Thanks for the report and sorry for the trouble!

I think I know the issue and while I don't know if I'll be able to actually run tests on FreeBSD on CI, I'll add a check to at least compile the code for it.

**Comment by anuraaga** [comment]
Thanks again for the report @yurivict - I have released 1.6.0 including the fix for this issue. Sorry for the trouble.

**Comment by yurivict** [comment]
@anuraaga 

Thank you for fixing it so quick!
